### PR TITLE
fix: ignore parsing helm templates 

### PR DESCRIPTION
**Pull request by mortada-codes**
This PR...

## Changes

- added a check function to check if the resource is supported

## Fixes

- The app was crashing when opening a helm charts template with variables `{{ . }}` 

## How to test it

- open a helm charts project 
- should not see a helm chart templates

## Screenshots

- 
Screen Shot 2022-01-18 at 1 44 37 PM

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: src/redux/services/helm.ts [hunk 1]
```diff
@@ -28,6 +29,15 @@ export function isHelmChartFolder(files: string[]) {
   return files.indexOf('Chart.yaml') !== -1 && files.indexOf('values.yaml') !== -1;
 }
 
+/**
+ * check if the k8sResource is supported
+ * @param resource
+ * @returns @boolean
+ */
+function isSupportedHelmResource(resource: K8sResource): boolean {
+  return !JSON.stringify(resource.content).includes('{"{');
```
Comment (olensmar) [inline src/redux/services/helm.ts:38]: you can use resource.text instead - which won't require you to stringify. Also perhaps we should use a regex to check for a matching closing pair? i.e. '{{.*}}'

Comment (mortada-codes) [reply src/redux/services/helm.ts]: resolved

File: src/redux/services/helm.ts [hunk 2]
```diff
@@ -35,7 +35,9 @@ export function isHelmChartFolder(files: string[]) {
  * @returns @boolean
  */
 function isSupportedHelmResource(resource: K8sResource): boolean {
-  return !JSON.stringify(resource.content).includes('{"{');
+  const helmVariableRegex = /{"{/g;
```
Comment (olensmar) [inline src/redux/services/helm.ts:38]: shouldn't this just be `{{.*}}` ?

Comment (mortada-codes) [reply src/redux/services/helm.ts:38]: yeah, I am fixing it

Comment (mortada-codes) [reply src/redux/services/helm.ts]: it is good now

**Review by devcatalin — APPROVED** [review]
LGTM

**Review by olensmar — APPROVED** [review]

**Comment by mortada-codes** [comment]
should i update the test cases before merge?

**Comment by olensmar** [comment]
> should i update the test cases before merge?

not sure - do they need updating?

**Comment by olensmar** [comment]
then you should update them accordingly!

**Review by olensmar — APPROVED** [review]

**Pull request merged**